### ISSUE TITLE
Zeroconf browser omits interface for link-local IPs

### ISSUE THREAD

**Issue by ilya-fedin** [xbmc/xbmc#23623]
### Describe the bug
Here is a clear and concise description of what the problem is:

I want to add a samba share via Zeroconf browser but it returns an "Invalid arument" error. Looking in the log, I see the following lines:
```
2023-08-16 06:57:26.592 T:753     error <general>: Unable to lookup host: 'fe80::1dcd:579:1191:1681'
2023-08-16 06:57:26.592 T:753     error <general>: SMBDirectory->GetDirectory: Unable to open directory : 'smb://fe80%3a%3a1dcd%3a579%3a1191%3a1681:445/'
                                                   unix_err:'16' error : 'Invalid argument'
2023-08-16 06:57:27.778 T:753     error <general>: GetDirectory - Error getting smb://[fe80::1dcd:579:1191:1681]:445/
2023-08-16 06:57:27.779 T:753     error <general>: GetDirectory - Error getting zeroconf://_smb._tcp.%40local%40NIXOS/
2023-08-16 06:57:27.779 T:753     error <general>: CGUIDialogFileBrowser::GetDirectory(zeroconf://_smb._tcp.%40local%40NIXOS/) failed
```
Looks like it tries to use an IPv6 link-local address (why?) but doesn't specify the interface. That is, `fe80::1dcd:579:1191:1681` doesn't ping but `fe80::1dcd:579:1191:1681%end0` pings just fine and even works via smbclient!

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Samba shares are accessible via zeroconf browser

## Actual Behavior

Invalid argument

## Possible Fix

This issue seem to give some highlights on how to process avahi's result right (particularly the linked commit): https://github.com/lathiat/avahi/issues/110

### To Reproduce

1. Set up a samba share on another machine with zeroconf support (seem to be a samba build option and depends on avahi configuration, seem to work out of the box on Arch Linux after samba & avahi installation)
2. Try to access it via Kodi's Zeroconf browser

### Debuglog

The debuglog can be found here:
[kodi.log](https://github.com/xbmc/xbmc/files/12354776/kodi.log)

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Arch Linux ARM
 - Kodi version: 20.1 (20.1.0)

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by becarpenter** [comment]
@ilya-fedin, you might be interested by new comments at https://github.com/avahi/avahi/issues/110

**Comment by ilya-fedin** [comment]
.

**Comment by becarpenter** [comment]
No, it isn't stale, it's just an open bug. avahi doesn't support link local zone IDs properly and linux doesn't support the default zone ID.

**Comment by ilya-fedin** [comment]
Sadly, still the same error:
```
2026-03-21 11:19:44.846 T:3752    error <general>: SMBDirectory->GetDirectory: Unable to open directory : 'smb://fe80%3a%3a3cb8%3a14ff%3afe3d%3ab88a:445/'
                                                   unix_err:'16' error : 'Invalid argument'
2026-03-21 11:19:48.652 T:3752    error <general>: GetDirectory - Error getting smb://[fe80::3cb8:14ff:fe3d:b88a]:445/
2026-03-21 11:19:48.652 T:3752    error <general>: GetDirectory - Error getting zeroconf://_smb._tcp.%40local%40BEELINK-SER5/
2026-03-21 11:19:48.652 T:3752    error <general>: CGUIDialogFileBrowser::GetDirectory(zeroconf://_smb._tcp.%40local%40BEELINK-SER5/) failed
```

**Comment by malard** [comment]
*This reply was written and posted by Claude on @malard's behalf.*

Confirmed on current `master`, and the line is identifiable.

`CZeroconfBrowserAvahi::resolveCallback` receives the interface index as its second parameter, `AvahiIfIndex interface` (`xbmc/platform/linux/network/zeroconf/ZeroconfBrowserAvahi.cpp:306-308`), and then never uses it. The address is stringified with `avahi_address_snprint(a, sizeof(a), address)` and stored as-is (`:341-342`), so a link-local result becomes `fe80::…` with no `%iface` suffix - unroutable, which is the "Invalid argument" you see.

The avahi issue you linked is the right reference: the interface has to be appended as a zone index for link-local addresses specifically. Everything needed to do that is already in scope at that call site.

**Related but not the same bug:** xbmc/xbmc#25957 is the identical *symptom* from a different producer - `CNetworkBase::GetIpStr` (`xbmc/network/Network.cpp:505-507`) formats only `sin6_addr` and discards `sin6_scope_id`, so link-local addresses resolved through `CDNSNameCache::Lookup` lose their zone index too. Two separate code sites, same defect class. Whoever fixes one should look at the other; they are being cross-linked rather than merged because neither fix covers the other.

Labelling `Component: Network`, `Platform: Linux`, `Triage: Confirmed`.

**Comment by ilya-fedin** [comment]
Nice, hope it will finally get fixed, seeing the typical way of discovering samba services used on Linux still not working with Kodi is very surprising (Windows Explorer uses WSD, Linux file managers use mDNS).
